### PR TITLE
Chart refactor

### PR DESCRIPTION
**Pull request by benma**
This is still not enough to pass the complexity linter :see_no_evil: But it the function is shorter and the code a bit more organized so I opened this PR.

### CODE REVIEW

File: backend/chart.go [hunk 1]
```diff
@@ -36,20 +36,32 @@ func (backend *Backend) allCoinCodes() []string {
...
+}
 
+// Chart has all data needed to show a time-based chart of their assets to the user.
+type Chart struct {
+	DataMissing bool         `json:"chartDataMissing"`
+	DataDaily   []ChartEntry `json:"chartDataDaily"`
+	DataHourly  []ChartEntry `json:"chartDataHourly"`
+	Fiat        string       `json:"chartFiat"`
+	Total       *float64     `json:"chartTotal"`
+	// only valid is DataMissing is false
```
Comment (x1ddos) [inline backend/chart.go:52]: "valid **if** ..."

File: backend/chart.go [hunk 2]
```diff
@@ -36,20 +36,32 @@ func (backend *Backend) allCoinCodes() []string {
...
+	Value float64 `json:"value"`
+}
 
+// Chart has all data needed to show a time-based chart of their assets to the user.
+type Chart struct {
+	DataMissing bool         `json:"chartDataMissing"`
+	DataDaily   []ChartEntry `json:"chartDataDaily"`
+	DataHourly  []ChartEntry `json:"chartDataHourly"`
+	Fiat        string       `json:"chartFiat"`
+	Total       *float64     `json:"chartTotal"`
```
Comment (x1ddos) [inline backend/chart.go:51]: might as well remove the pointer here? there's `DataMissing` for that. or is it also valid when data is missing?

File: backend/chart.go [hunk 3]
```diff
@@ -188,30 +216,8 @@ func (backend *Backend) ChartData() (*Chart, error) {
 			return nil, err
 		}
 
-		addChartData := func(coinCode coin.Code, timeseries []accounts.TimeseriesEntry, chartEntries map[int64]ChartEntry) {
```
Comment (x1ddos) [inline backend/chart.go]: Nice. I would also move `toSortedSlice` out. It really isn't a big deal - you see somehing "sorted", you'll know immediately what it does without even looking at its code. But moving `toSortedSlice` out makes it easier to read the rest of the important logic in this method.

**Review by x1ddos — APPROVED** [review]
LGTM

File: backend/chart.go (hunk 1 above)
Comment (benma) [reply backend/chart.go:52]: :see_no_evil:

File: backend/chart.go (hunk 2 above)
Comment (benma) [reply backend/chart.go:51]: yeah, it is independent of DataMissing, but can also be missing :joy:  I will add comments to each field.

**Pull request merged**